frontend/src/components/nodes/ValidatorsList.tsx

```diff
@@ -0,0 +1,74 @@
+import BN from "bn.js";
+import React from "react";
+
+import * as N from "../../libraries/explorer-wamp/nodes";
+
+import ValidatorRow from "./ValidatorRow";
+
+interface Props {
+  validators: any;
+  pages: any;
+  cellCount: number;
+  validatorType: string;
+}
+
+class ValidatorsList extends React.PureComponent<Props> {
+  calculateStake = (nodeIndex: number, totalStake: BN) => {
+    let total = new BN(0);
+    let networkHolderIndex = [];
+    for (let i = 0; i <= nodeIndex; i++) {
+      total = total.add(new BN(this.props.validators[i].stake));
+
+      if (total.gt(totalStake.div(new BN(3)))) {
```

Comment:
Precompute the `totalStake.divn(3)` in a const before the loop, so you don't need to re-compute it over and over again in the loop.